presto-hive/src/main/java/io/prestosql/plugin/hive/HiveSplitSource.java

```diff
@@ -414,6 +456,70 @@ void fail(Throwable e)
         return toCompletableFuture(transform);
     }
 
+    private List<ConnectorSplit> dynamicallyFilterSplits(List<ConnectorSplit> splits)
+    {
+        TupleDomain<HiveColumnHandle> runtimeTupleDomain = dynamicPredicates.get();
+        if (runtimeTupleDomain.isAll()) {
+            return splits;
+        }
+        else if (runtimeTupleDomain.isNone()) {
+            return ImmutableList.of();
+        }
+        Map<String, Domain> domains = runtimeTupleDomain.getDomains().get()
+                .entrySet().stream()
+                .collect(Collectors.toMap(entry -> entry.getKey().getName(), Map.Entry::getValue));
+        return splits.stream()
+                .map(HiveSplit.class::cast)
+                .filter(hiveSplit -> !isPartitionPruned(domains, hiveSplit.getPartitionKeys()))
+                .collect(toImmutableList());
+    }
+
+    private boolean isPartitionPruned(Map<String, Domain> domains, List<HivePartitionKey> partitionKeys)
+    {
+        boolean partitionPruned = false;
+        for (HivePartitionKey partitionKey : partitionKeys) {
+            Domain allowedDomain = domains.get(partitionKey.getName());
+            if (allowedDomain != null && !allowedDomain.isAll()) {
+                if (!isHiveNull(partitionKey.getValue().getBytes())) {
+                    Type type = partitionKey.getHiveType().getType(typeManager);
+                    NullableValue partitionValue;
+                    try {
+                        partitionValue = parsePartitionValue(partitionKey.getName(), partitionKey.getValue(), type, timeZone);
+                    }
+                    catch (PrestoException e) {
+                        // if the type is not supported, skip pruning for that partition
+                        if (e.getErrorCode().equals(NOT_SUPPORTED.toErrorCode())) {
+                            log.warn("Type %s is not supported for Dynamic Partition Pruning", type);
```

Comment:
This was put in from developer point of view, so that we had some way to know that this problem occurred without failing the query for the user.
Since this exception handling itself is unnecessary, i'll get rid of it.